validatorobserver/validator_observer.go

```diff
@@ -0,0 +1,197 @@
+package validatorobserver
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/Ethernal-Tech/apex-bridge/common"
+	"github.com/Ethernal-Tech/apex-bridge/contractbinding"
+	"github.com/Ethernal-Tech/apex-bridge/eth"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/hashicorp/go-hclog"
+)
+
+type ValidatorSetObserver struct {
+	validatorSetStream  chan *Validators
+	validators          Validators
+	validatorSetPending bool
+	bridgeSmartContract eth.IBridgeSmartContract
+	logger              hclog.Logger
+}
+
+type ValidatorsChainData struct {
+	Keys       []eth.ValidatorChainData
+	SlotNumber uint64
+}
+
+type Validators struct {
```

Comment:
This should be name as `ValidatorsPerChain` and probably just be

`type ValidatorsPerChain map[string]ValidatorsChainData`